test/functorch/test_aotdispatch.py

```diff
@@ -2249,6 +2266,67 @@ def fn(p, x):
             aot_export_joint_simple(fn, [mod.p, inp], trace_joint=True)
             aot_export_module(mod, [inp], trace_joint=False)
 
+    def test_aot_export_functionalized_asserts(self):
+        def fn(p, x):
+            b = x.sin()
+            x0 = torch.ops.aten.select.int(x, 0, 0)
+            eq = torch.ops.aten.eq.Scalar(x0, 3)
+            torch.ops.aten._assert_async.msg(eq, "test assertion error")
+            return (x.cos() + b,)
+
+        mod = TestMod(fn)
+        inp = torch.tensor([3, 2, 1])
+
+        with patch(
+            "functorch.compile.config.functionalize_assertion_ops", True
+        ), self.assertRaisesRegex(
+            AssertionError,
+            "Cannot functionalize assertion ops when trace joint is enabled",
+        ):
+            aot_export_module(mod, [inp], trace_joint=True)
+
+        with patch("functorch.compile.config.functionalize_assertion_ops", True), patch(
+            "functorch.compile.config.functionalize_rng_ops", True
+        ), self.assertRaisesRegex(
+            AssertionError,
+            "Cannot functionalize assertion ops when RNG functionalization is enabled",
+        ):
+            aot_export_module(mod, [inp], trace_joint=False)
+
+        with patch("functorch.compile.config.functionalize_assertion_ops", True), patch(
+            "functorch.compile.config.functionalize_rng_ops", False
+        ):
+            fx_gm, signature = aot_export_module(mod, [inp], trace_joint=False)
+            self.assertExpectedInline(fx_gm.print_readable(print_output=False), """\
+class <lambda>(torch.nn.Module):
+    def forward(self, arg0_1: f32[2], arg1_1: i64[3]):
+        # No stacktrace found for following nodes
+        sin: f32[3] = torch.ops.aten.sin.default(arg1_1)
+        select: i64[] = torch.ops.aten.select.int(arg1_1, 0, 0)
+        eq: b8[] = torch.ops.aten.eq.Scalar(select, 3);  select = None
+        _make_dep_token: f32[] = torch.ops.aten._make_dep_token.default()
+        _functional_assert_async: f32[] = torch.ops.aten._functional_assert_async.msg(eq, 'test assertion error', _make_dep_token);  eq = _make_dep_token = None
```

Comment:
it might be worth including two asserts in the traced function, just so you can confirm from the test that the dependency exists in the graph and is being created properly.